core/src/tests/compat_old_conversion.cc

```diff
@@ -0,0 +1,393 @@
+                                            dwBufCharsNeeded * sizeof(wchar_t));
+
+    /* Get directory into own buffer as it may either return c:\... or
+     * \\?\C:\.... */
+    if (IsPathSeparator(pwszCurDirBuf[0]) && IsPathSeparator(pwszCurDirBuf[1])
+        && pwszCurDirBuf[2] == '?' && IsPathSeparator(pwszCurDirBuf[3])) {
+      // Copy complete string
+      wcscpy(pwszBuf, pwszCurDirBuf);
+    } else {
+      // Append path
+      wcscat(pwszBuf, pwszCurDirBuf);
+    }
+
+    nParseOffset = wcslen((LPCWSTR)pwszBuf);
+
+    // Check if path ends with backslash, if not, add one
+    if (!IsPathSeparator(pwszBuf[nParseOffset - 1])) {
+      wcscat(pwszBuf, L"\\");
+      nParseOffset++;
+    }
+  }
+
+  wchar_t* win32_name = &pwszBuf[nParseOffset];
+  wchar_t* name_start = name;
+  wchar_t previous_char = 0;
+  wchar_t next_char = 0;
+
+  while (*name) {
+    /* Check for Unix separator and convert to Win32, eliminating duplicate
+     * separators. */
+    if (IsPathSeparator(*name)) {
+      // Don't add a trailing slash.
+      next_char = *(name + 1);
+      if (previous_char != ':' && next_char == '\0') {
+        name++;
+        continue;
+      }
+      previous_char = '\\';
+      *win32_name++ = '\\'; /* convert char */
+
+      /* Eliminate consecutive slashes, but not at the start so that \\.\ still
+       * works. */
+      if (name_start != name && IsPathSeparator(name[1])) { name++; }
+    } else {
+      previous_char = *name;
+      *win32_name++ = *name; /* copy character */
+    }
+    name++;
+  }
+
+  // NULL Terminate string
+  *win32_name = 0;
+
+  /* Here we convert to VSS specific file name which can get longer because VSS
+   * will make something like
+   * \\\\?\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy1\\bareos\\uninstall.exe
+   * from c:\bareos\uninstall.exe */
+  thread_vss_path_convert* tvpc = Win32GetPathConvert();
+  if (tvpc) {
+    // Is output buffer large enough?
+    pwszBuf = (wchar_t*)CheckPoolMemorySize(
+        (POOLMEM*)pwszBuf, (dwBufCharsNeeded + MAX_PATH) * sizeof(wchar_t));
+    // Create temp. buffer
+    wchar_t* pszBuf = (wchar_t*)GetPoolMemory(PM_FNAME);
+    pszBuf = (wchar_t*)CheckPoolMemorySize(
+        (POOLMEM*)pszBuf, (dwBufCharsNeeded + MAX_PATH) * sizeof(wchar_t));
+    if (bAddPrefix) {
+      nParseOffset = 4;
+    } else {
+      nParseOffset = 0;
+    }
+
+    wcsncpy(pszBuf, &pwszBuf[nParseOffset], wcslen(pwszBuf) + 1 - nParseOffset);
+    tvpc->pPathConvertW(pszBuf, pwszBuf, dwBufCharsNeeded + MAX_PATH);
+    FreePoolMemory((POOLMEM*)pszBuf);
+  }
+
+  FreePoolMemory(pszUCSPath);
+  FreePoolMemory((POOLMEM*)pwszCurDirBuf);
+
+  Dmsg1(debuglevel, "Leave make_wchar_win32_path=%s\n", pwszBuf);
+  return (POOLMEM*)pwszBuf;
+}
+
+int make_win32_path_UTF8_2_wchar(POOLMEM*& pszUCS,
+                                 const char* pszUTF,
+                                 BOOL* pBIsRawPath /*= NULL*/)
+{
+  int nRet;
+  // FUTURE NOTE: no caching inside the tests!
```

Comment:
Is the following block still required?

Reply:
No.  Ill remove them.  This is just a note that I changed the "old" code slightly to make it suitable for regression testing.  But we do not need the actual code for that; just the comment is enough.